docs/narratives/build_backends.md

```diff
@@ -0,0 +1,217 @@
+a matrix should be straightforward.
+
+For wheels distributed to PyPI, an easy way to do this may be to leverage what
+is already being done for building a set of wheels for diffent platforms and
+Python versions, which is typically to use a build matrix in GitHub Actions:
+
+```yaml
+jobs:
+  build_wheel_variants:
+    name: Wheel, ${{ matrix.python[0] }}-${{ matrix.buildplat[1] }}
+      ${{ matrix.buildplat[2] }} ${{ matrix.buildplat[3] }}
+      ${{ matrix.buildplat[4] }}
+    runs-on: ${{ matrix.buildplat[0] }}
+
+    strategy:
+      matrix:
+        buildplat:
+        - [ubuntu-22.04, manylinux, x86_64, "", ""]
+        - [ubuntu-22.04, musllinux, x86_64, "", ""]
+        - [ubuntu-22.04, manylinux, x86_64, "psabi-v3", ""]
+        - [ubuntu-22.04, manylinux, x86_64, "psabi-v3", "mkl"]
+        python: ["cp312", "cp313"]
+
+    steps:
+      - name: Checkout source of pacakge
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: Setup Python interpreter to run cibuildwheel
+        uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5.3.0
+        with:
+          python-version: 3.13
+
+      - name: Handle variant builds
+        run: |
+          # psabi (x86-64 SIMD) variants
+          if [[ ${{ matrix.buildplat[2] }} != '' ]]; then
+            # Will still include dispatching for AVX512 features as well, it just
+            # increases the baseline features that are required.
+            export VARIANT_CONFIG=\"variant-name=${{ matrix.buildplat[2] }} cpu-baseline=AVX2,FMA3\"
+          fi
+
+          # BLAS variant
+          if [[ ${{ matrix.buildplat[3] }} == 'mkl' ]]; then
+            python -m pip install mkl
+            export VARIANT_CONFIG=\"variant-name=mkl setup-args=-Dblas=mkl\ $VARIANT_CONFIG"
+          fi
+          echo CIBW_CONFIG_SETTINGS=$VARIANT_CONFIG >> "$GITHUB_ENV"
+
+      - name: Build wheels
+        uses: pypa/cibuildwheel@7940a4c0e76eb2030e473a5f864f291f63ee879b # v2.21.3
+        env:
+          CIBW_BUILD: ${{ matrix.python[0] }}-${{ matrix.buildplat[1] }}*
+          CIBW_ARCHS: ${{ matrix.buildplat[2] }}
+          TODO: pass on build variants!
+
+      - uses: actions/upload-artifact@b4b15b8c7c6ac21ea08fcf65892d2ee8f75cf882 # v4.4.3
+        with:
+          path: ./wheelhouse/*.whl
+          name: ${{ matrix.python[0] }}-${{ matrix.buildplat[1] }}
+            ${{ matrix.buildplat[2] }} ${{ matrix.buildplat[3] }}
+            ${{ matrix.buildplat[4] }}
+```
+
+This should yield four wheels named like (plus four more with `cp312` instead of `cp313`):
+```
+numpy-2.2.4-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
+numpy-2.2.4-cp313-cp313-manylinux_2_17_x86_64.musllinux2014_x86_64.whl
+numpy-2.2.4-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64_h03981ea44_psabi3.whl
+numpy-2.2.4-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64_h0f4317532_psabi3_mkl.whl
```

Comment:
You think it's better for the `variant_hash+name` to be at the end ?

The current design has it slightly different: https://mockhouse.wheelnext.dev/pep-xxx-variants/dummy-project/

I like your idea of `+alias` I would use the `+` character :)

Reply:
Let's defer that to gh-5? @mgorny has already thought about it harder than I have. Here the only key bit is we need the strings in addition to the hash. I don't mind too much where it goes, at least for the purposes of this build backend focused exercise.

Reply:
Agreed - it fundamentally doesn't change anything

Reply:
Yeah, let's discuss it there. However, note that as it is stated now, it will be interpreted as "platform tag" of (`manylinux_2_17_x86_64` OR `manylinux2014_x86_64_h03981ea44_psabi3`), so a non-variant-enabled PM will consider all versions suitable for the former platform.

Reply:
Good catch. I changed this to match https://github.com/wheelnext/mockhouse/pull/4 now, so I'll resolve this comment.